### ISSUE TITLE
 'SlurmComputationEngine' object has no attribute 'all'

### ISSUE THREAD

**Issue by wittawatj** [karlnapf/independent-jobs#2]
According to the log, probably some of the submitted jobs were killed by Slurm. The computation engine seems to have submitted again. But then, I am not sure why I got

```
AttributeError: 'SlurmComputationEngine' object has no attribute 'all'
```

Full log

```
INFO: 2015-12-09 21:56:03,634: BatchClusterComputationEngine._wait_until_n_unfinished(): Waiting for e1_8b4b6ea0-b273-47df-83ed-807
a1b783db2 and 219 other jobs
INFO: 2015-12-09 21:56:29,261: BatchClusterComputationEngine._wait_until_n_unfinished(): e1_8b4b6ea0-b273-47df-83ed-807a1b783db2 ex
ceeded maximum waiting time of 3540h
INFO: 2015-12-09 21:56:29,261: BatchClusterComputationEngine._resubmit(): Re-submitting under name e1_00dbeb1f-91d9-4ba2-b2dd-d234e
8993689
Traceback (most recent call last):
  File "freqopttest/ex/ex1_power_vs_n.py", line 199, in <module>
    main()
  File "freqopttest/ex/ex1_power_vs_n.py", line 168, in main
    engine.wait_for_all()
  File "/nfs/nhome/live/wittawat/git/independent-jobs/independent_jobs/engines/BatchClusterComputationEngine.py", line 279, in wait
_for_all
    self._wait_until_n_unfinished(0)
  File "/nfs/nhome/live/wittawat/git/independent-jobs/independent_jobs/engines/BatchClusterComputationEngine.py", line 274, in _wai
t_until_n_unfinished
    self._resubmit(job_name)
  File "/nfs/nhome/live/wittawat/git/independent-jobs/independent_jobs/engines/BatchClusterComputationEngine.py", line 223, in _res
ubmit
    for i in range(len(self.all)):
AttributeError: 'SlurmComputationEngine' object has no attribute 'all'
```

**Comment by karlnapf** [comment]
That is a bug. I wil look into it when I am back.

**Comment by karlnapf** [comment]
Fixed in a4e5bf09e1b5d86dc04d5d9f090f5404914b5d6f
